plugins/groups_manager.py

```diff
@@ -0,0 +1,307 @@
+# Red Hat, Inc.
+#
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import argparse
+import gzip
+import libcomps
+import os
+import re
+import shutil
+import tempfile
+
+from dnfpluginscore import _, logger
+import dnf
+import dnf.cli
+
+
+RE_GROUP_ID_VALID = '-a-z0-9_.:'
+RE_GROUP_ID = re.compile(r'^[{}]+$'.format(RE_GROUP_ID_VALID))
+RE_LANG = re.compile(r'^[-a-zA-Z0-9_.@]+$')
+COMPS_XML_OPTIONS = {
+    'default_explicit': True,
+    'uservisible_explicit': True,
+    'empty_groups': True}
+
+
+def group_id_type(value):
+    '''group id validator'''
+    if not RE_GROUP_ID.match(value):
+        raise argparse.ArgumentTypeError(_('Invalid group id'))
+    return value
+
+
+def translation_type(value):
+    '''translated texts validator'''
+    data = value.split(':', 2)
+    if len(data) != 2:
+        raise argparse.ArgumentTypeError(
+            _("Invalid translated data, should be in form 'lang:text'"))
+    lang, text = data
+    if not RE_LANG.match(lang):
+        raise argparse.ArgumentTypeError(_('Invalid/empty language for translated data'))
+    return lang, text
+
+
+def text_to_id(text):
+    '''generate group id based on its name'''
+    group_id = text.lower()
+    group_id = re.sub('[^{}]'.format(RE_GROUP_ID_VALID), '', group_id)
+    if not group_id:
+        raise dnf.cli.CliError(
+            _("Can't generate group id from '{}'. Please specify group id using --id.").format(
+                text))
+    return group_id
+
+
+@dnf.plugin.register_command
+class GroupsManagerCommand(dnf.cli.Command):
+    aliases = ('groups-manager',)
+    summary = _('create and edit groups metadata file')
+
+    def __init__(self, cli):
+        super(GroupsManagerCommand, self).__init__(cli)
+        self.comps = libcomps.Comps()
+
+    @staticmethod
+    def set_argparser(parser):
+        # input / output options
+        parser.add_argument('--load', action='append', default=[],
+                            metavar='COMPS.XML',
+                            help=_('load groups metadata from file'))
+        parser.add_argument('--save', action='append', default=[],
+                            metavar='COMPS.XML',
+                            help=_('save groups metadata to file'))
+        parser.add_argument('--merge', metavar='COMPS.XML',
+                            help=_('load and save groups metadata to file'))
+        parser.add_argument('--print', action='store_true', default=False,
+                            help=_('print the result metadata to stdout'))
+        # group options
+        parser.add_argument('--id', type=group_id_type,
+                            help=_('group id'))
+        parser.add_argument('-n', '--name', help=_('group name'))
+        parser.add_argument('--description',
+                            help=_('group description'))
+        parser.add_argument('--display-order', type=int,
+                            help=_('sort order override'))
```

Comment:
I feel the help text doesn't really describe the option - at first I thought it's influencing the output in some way, but it is actually setting a group's attribute. Saying simply "group display order" seems better.